examples/stringification/main.cpp

```diff
@@ -0,0 +1,491 @@
+    
+    CHECK_EQ(toString(static_cast<unsigned long long int>(0ull)), "0");
+    CHECK_EQ(toString(static_cast<unsigned long long int>(13324909658562760708ull)), "13324909658562760708");
+}
+
+TEST_CASE("Boolean_type")
+{
+    CHECK_EQ(toString(true), "true");
+    CHECK_EQ(toString(false), "false");
+}
+
+TEST_CASE("Standard_character_types")
+{
+    CHECK_EQ(toString(static_cast<signed char>('A')), "65");
+    CHECK_EQ(toString(static_cast<unsigned char>('B')), "66");
+    CHECK_EQ(toString(static_cast<char>('C')), "67");
+}
+
+// String representation is lost in c++20 for whatever reason
+TEST_CASE("Wide_character_types")
+{
+#ifdef ENABLE_BROKEN_TESTS
+#   if CXX_STANDARD >= 20
+    CHECK_EQ(toString(static_cast<wchar_t>(L'A')), "{?}");
+    CHECK_EQ(toString(static_cast<char16_t>(L'B')), "{?}");
+    CHECK_EQ(toString(static_cast<char32_t>(L'C')), "{?}");
+#   else
+    CHECK_EQ(toString(static_cast<wchar_t>(L'A')), "65");
+    CHECK_EQ(toString(static_cast<char16_t>(L'B')), "66");
+    CHECK_EQ(toString(static_cast<char32_t>(L'C')), "67");
+#   endif
+#else
+    add_stub_asserts(3);
+#endif
+}
+
+TEST_CASE("Floating_point_types_zero")
+{
+    CHECK_EQ(toString(static_cast<float>(0.0f)), "0.0f");
+    CHECK_EQ(toString(static_cast<double>(0.0)), "0.0");
+    CHECK_EQ(toString(static_cast<long double>(0.0L)), "0.0");
+}
+
+// Output format is quite fragile due to absence of output precision
+// For example 904.1f may be printed as 904.09998f
+// I have no idea how to test it reliably due to this fact
+TEST_CASE("Floating_point_types__non_zero" * may_fail())
+{
+    CHECK_EQ(toString(static_cast<float>(-10.0f)), "-10.0f");
+    CHECK_EQ(toString(static_cast<float>(-1.1f)), "-1.1f");
+    CHECK_EQ(toString(static_cast<float>(105.1f)), "105.1f");
+    CHECK_EQ(toString(static_cast<float>(51425.0f)), "51425.0f");
+    
+    CHECK_EQ(toString(static_cast<double>(-10.0)), "-10.0");
+    CHECK_EQ(toString(static_cast<double>(-1.1)), "-1.1");
+    CHECK_EQ(toString(static_cast<double>(105.1)), "105.1");
+    CHECK_EQ(toString(static_cast<double>(51425.0)), "51425.0");
+    
+    CHECK_EQ(toString(static_cast<long double>(-10.0L)), "-10.0");
+    CHECK_EQ(toString(static_cast<long double>(-1.1L)), "-1.1");
+    CHECK_EQ(toString(static_cast<long double>(15.1L)), "15.1");
+    CHECK_EQ(toString(static_cast<long double>(51425.0L)), "51425.0");
+}
+
+TEST_CASE("Pointer_types")
+{
+#if SIZEOF_VOID_P == 4
+#   define POINTER_PREFIX "0x"
+#elif SIZEOF_VOID_P == 8
+#   define POINTER_PREFIX "0x00000000"
+#endif
+    
+    // doctest overloads toString() for pointer types. So string doesn't match ostreamed result
+    // NULL is handled separatly somewhat
+    // other value representation has fixed size string representation depending on sizeof(void*)
+    CHECK_EQ(toString(static_cast<int*>(nullptr)), "NULL");
+    CHECK_EQ(toString(reinterpret_cast<int*>(0xABADF00D)), POINTER_PREFIX "abadf00d");
```

Comment:
Correct me if I'm wrong, but I'm fairly certain stringification of pointers is implementation defined (comes down to `printf` with the `%p` specifier). Doctest only handles `nullptr`s seperately. We could implement pointer stringification implementation-independently but I'm honestly not sure if we should bother. (E.g. it would require template metaprogramming to get the amount of padding 0s needed.)